homeassistant/components/zwave_js/sensor.py

```diff
@@ -789,19 +767,13 @@ def __init__(
             alternate_value_name=self.info.primary_value.property_name,
             additional_info=[property_key_name] if property_key_name else None,
         )
-
-    @property  # type: ignore[override]
-    def device_class(self) -> SensorDeviceClass | None:
-        """Return sensor device class."""
-        # mypy doesn't know about fget: https://github.com/python/mypy/issues/6185
-        if (device_class := ZwaveSensor.device_class.fget(self)) is not None:  # type: ignore[attr-defined]
-            return device_class  # type: ignore[no-any-return]
         if (
-            self._primary_value.configuration_value_type
+            self.entity_description.device_class is None
+            and not hasattr(self, "_attr_device_class")
```

Comment:
I guess we don't have to check this since `_attr_device_class` can only be `SensorDeviceClass.ENUM` above

Reply:
This really needs a test as I feel uncomfortable changing this

Reply:
Its a bit more heavy but I think these are rare so it probably doesn't matter as this seems safe
```diff
diff --git a/homeassistant/components/zwave_js/sensor.py b/homeassistant/components/zwave_js/sensor.py
index 6059eeee524..eaee8020dd3 100644
--- a/homeassistant/components/zwave_js/sensor.py
+++ b/homeassistant/components/zwave_js/sensor.py
@@ -19,7 +19,7 @@ from zwave_js_server.model.node import Node as ZwaveNode
 from zwave_js_server.model.node.statistics import NodeStatisticsDataType
 from zwave_js_server.model.value import ConfigurationValue, ConfigurationValueType
 from zwave_js_server.util.command_class.meter import get_meter_type
-
+from dataclasses import replace
 from homeassistant.components.sensor import (
     DOMAIN as SENSOR_DOMAIN,
     SensorDeviceClass,
@@ -730,7 +730,10 @@ class ZWaveListSensor(ZwaveSensor):
             additional_info=[self.info.primary_value.property_key_name],
         )
         if self.info.primary_value.metadata.states:
-            self._attr_device_class = SensorDeviceClass.ENUM
+            if self.entity_description.device_class != SensorDeviceClass.ENUM:
+                self.entity_description = replace(
+                    self.entity_description, device_class=SensorDeviceClass.ENUM
+                )
             self._attr_options = list(info.primary_value.metadata.states.values())
 
     @property
@@ -768,12 +771,13 @@ class ZWaveConfigParameterSensor(ZWaveListSensor):
             additional_info=[property_key_name] if property_key_name else None,
         )
         if (
-            self.entity_description.device_class is None
-            and not hasattr(self, "_attr_device_class")
-            and self._primary_value.configuration_value_type
+            self._primary_value.configuration_value_type
             == ConfigurationValueType.ENUMERATED
+            and self.entity_description.device_class != SensorDeviceClass.ENUM
         ):
-            self._attr_device_class = SensorDeviceClass.ENUM
+            self.entity_description = replace(
+                self.entity_description, device_class=SensorDeviceClass.ENUM
+            )
 
     @property
     def extra_state_attributes(self) -> dict[str, str] | None:
```